cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -159,30 +155,21 @@ private void addShuffledGoalsForPrecomputing(Set<List<Goal>> shuffledGoals) {
       boolean isOriginalOrder = true;
       for (int i = 0; i < shuffledGoalList.size(); i++) {
         String shuffledGoal = shuffledGoalList.get(i).name();
-        if (!shuffledGoal.equals(originalGoals.get(i).name())) {
+        if (!shuffledGoal.equals(_goalsByPriority.get(i).name())) {
           isOriginalOrder = false;
           break;
         }
       }
       if (!isOriginalOrder) {
-        addGoalsForPrecomputing(shuffledGoalList);
+        _goalByPriorityForPrecomputing.add(new ArrayList<>(shuffledGoalList));
```

Comment:
We used to create the copy of the `shuffledGoalList` in `addGoalsForPrecomputing`, which we now perform here to avoid overwriting the same shuffled list.